sys-utils/libcmt/Makefile

```diff
@@ -0,0 +1,191 @@
+# paths
+PREFIX = /usr
+TARGET_PREFIX ?= $(PREFIX)
+
+TOOLCHAIN_PREFIX ?= riscv64-linux-gnu-
+TARGET_CC := $(TOOLCHAIN_PREFIX)gcc
+TARGET_AR := $(TOOLCHAIN_PREFIX)ar
+TARGET_CFLAGS := -Wvla -O2 -g -Wall -pedantic -Wextra -Iinclude -Iinclude/libcmt/ioctl
+CFLAGS := -Wvla -O2 -g -Wall -pedantic -Wextra -fsanitize=address,undefined -Iinclude -Iinclude/libcmt/mock
+
+ALL := ioctl.build mock.build tools.build doc.build test.run
+all: $(ALL)
+
+#-------------------------------------------------------------------------------
+examples_SRC := \
+	doc/examples/io.c \
+	doc/examples/rollup.c
+
+examples_OBJDIR := .build/examples/
+examples_OBJ    := $(patsubst %.c,$(examples_OBJDIR)%.o,$(examples_SRC))
+
+$(examples_OBJ): $(examples_OBJDIR)%.o: %.c
+	@mkdir -p $(@D)
+	$(CC) $(CFLAGS) -MT $@ -MMD -MP -MF $(@:.o=.d) -c -o $@ $<
+
+# no need to link, just ensure examples build correctly
+examples.build: $(examples_OBJ)
+
+#-------------------------------------------------------------------------------
+ioctl_SRC := \
+	src/buf.c \
+	src/abi.c \
+	src/keccak.c \
+	src/merkle.c \
+	src/merkle-table.c \
+	src/rollup.c \
+	src/ioctl/io.c
+
+ioctl_OBJDIR    := build/ioctl/
+ioctl_OBJ       := $(patsubst %.c,$(ioctl_OBJDIR)%.o,$(ioctl_SRC))
+ioctl_LIB       := $(ioctl_OBJDIR)libcmt.a
```

Comment:
We also need to provide a stripped shared library, for programming languages that cannot link to C static libraries, but can use `dlopen` and have a C FFI.

Reply:
I would rather leave this to a separate PR.